controllers/deployment.go

```diff
@@ -186,9 +186,15 @@ func buildStatefulSets(master *v1alpha1.CDAPMaster, name string, services Servic
 			continue
 		}
 
-		c := serviceContainerSpec(ss, master, dataDir, s)
+		// Merge environment variables between service spec and master spec
+		env, err := mergeEnvVars(&master.Spec, ss)
```

Comment:
We pass master and service spec (ss) too the serviceContainerSpec function, we could move this merging step into serviceContainerSpec() and reduce duplication while creating STS spec and Deployment Spec.

Reply:
Fixed, thanks!